DOCS-1600: Add configure a fleet page (#2466)
viamrobotics/docs: docs/fleet/configure-a-fleet.md

---
title: "Use Fragments to Configure a Fleet of Machines"
linkTitle: "Configure a Fleet"
weight: 40
type: "docs"
description: Use fragments to configure many machines at the same time.
tags: ["fleet management", "cloud", "app"]
---

If you have multiple identical or similar machines, use a _fragment_ to configure all of the machines at the same time.

Fragments are a way of sharing and managing [machine configurations](/build/configure/) across multiple machines.

> When changes are made to the fragment, the Viam app automatically applies the changes to all machines that include the fragment in their config.

Reviewer: ```suggestion
When you or one of your collaborators changes the fragment, the Viam app automatically applies the changes to all machines that include the fragment in their config.
```

---
title: "Use Fragments to Configure a Fleet of Machines"
linkTitle: "Configure a Fleet"
weight: 40
type: "docs"
description: Use fragments to configure many machines at the same time.
tags: ["fleet management", "cloud", "app"]
---

If you have multiple identical or similar machines, use a _fragment_ to configure all of the machines at the same time.

> Fragments are a way of sharing and managing [machine configurations](/build/configure) across multiple machines.

Author: ```suggestion
Fragments are a way of sharing and managing [machine configurations](/build/configure/) across multiple machines.
```

Fragments are a way of sharing and managing identical [configuration](/build/configure/) files across multiple machines.
When changes are made to the fragment, those changes are automatically carried to all machines that include the fragment in their config.

If there are differences between your machines, you can use a fragment to quickly configure the {{< glossary_tooltip term_id="resource" text="resources" >}} that are the same between machines.
You can configure the differing resources separately, outside of the fragment.

## Create a fragment

Before you create a fragment, you'll need a configuration file.
Start by [configuring](/build/configure/) one of your identical machines on its **CONFIG** tab in the [Viam app](https://app.viam.com).
Now you're ready to share that config by creating a fragment.

To create your own private fragment, go to [app.viam.com/fragments](https://app.viam.com/fragments) or click on **Fragments** in the left navigation bar on the [FLEET page](https://app.viam.com/robots).

1. Enter a name for your new fragment and click **Add fragment**.
2. Add the JSON configuration in the config field.

>    You can get the JSON configuration from the [**CONFIG** tab](/build/configure/#the-config-tab) of your first machine by switching to **Raw JSON** mode.

Reviewer: that sounds good too
